config/routes.rb

```diff
@@ -0,0 +1,24 @@
+Rails.application.routes.draw do
+  devise_for :users
+  root to: 'recipes#public'
+
+  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+
+  # Defines the root path route ("/")
+  # root "articles#index"
+  get '/recipes', to: 'recipes#index', as: 'recipes'
+  get '/public_recipes', to: 'recipes#public', as: 'public_recipes'
+  get '/recipes/new', to: 'recipes#new', as: 'recipe_new'
+  post 'recipes', to: 'recipes#create', as: 'recipe_create'
+  delete 'recipes/:id/delete', to: 'recipes#destroy', as: 'recipe_delete'
+  get '/recipes/:id', to: 'recipes#show', as: 'recipe'
+  get '/recipes/:recipe_id/new_ingredient', to: 'recipe_foods#new', as: 'ingredient_new'
+  post 'recipes/:recipe_id', to: 'recipe_foods#create', as: 'ingredient_create'
+  get '/recipes/:recipe_id/update/:id', to: 'recipe_foods#edit', as: 'ingredient_edit'
+  put 'recipes/:recipe_id/update/:id', to: 'recipe_foods#update', as: 'ingredient_update'
+  delete 'recipes/:recipe_id/delete/:id', to: 'recipe_foods#destroy', as: 'ingredient_delete'
+  post '/recipes/:id/toggle', to: 'recipes#toggle', as: 'recipe_toggle'
+  get '/recipes/:recipe_id/shopping_list', to: 'recipe_foods#shopping_list', as: 'shopping_list'
```

Comment:
- [ ] [optional]It would have been nice if you used resources for your recipe routes , as you can see writing the routes takes time and difficult to scale your app in future.